internal/core/services/operation_manager/operation_manager.go

```diff
@@ -298,6 +304,16 @@ func (om *OperationManager) StartLeaseRenewGoRoutine(operationCtx context.Contex
 	}()
 }
 
+func (om *OperationManager) AppendOperationEventToExecutionHistory(ctx context.Context, op *operation.Operation, eventMessage string) error {
+	event := operation.OperationEvent{
+		CreatedAt: time.Now().UTC(),
+		Event:     eventMessage,
+	}
+	op.ExecutionHistory = append(op.ExecutionHistory, event)
+
+	return om.Storage.UpdateOperationExecutionHistory(ctx, op)
```

Comment:
I think this method is not supposed to break. What's your opinion about it?
Maybe trying to update storage fails, and we log ERROR, but the append method itself does not returns an error. I don't see a reason why appending should break something. I just imagine an error being returned for logging reasons, which the method itself could be responsible for doing